brave-spancollector-scribe/src/main/java/com/github/kristofa/brave/scribe/ScribeSpanCollectorNotificationsHandler.java

```diff
@@ -0,0 +1,35 @@
+package com.github.kristofa.brave.scribe;
+
+import com.twitter.zipkin.gen.Span;
+
+/**
+ * Monitor scribe span collector's performance by implementing reactions to these notifications,
+ * e.g. by updating some metrics in the metrics tool used in your application.
+ */
+public interface ScribeSpanCollectorNotificationsHandler {
+
+    /**
+     * Called when span queue size changes.
+     *
+     * @param currentQueueSize the current queue size.
+     */
+    void onQueueSizeChanged(int currentQueueSize);
+
+    /**
+     * Called when a span could not be added to the span queue because of it being full
+     * and so the span was rejected.
+     *
+     * @param rejectedSpan the rejected span.
+     */
+    void onSpanRejected(Span rejectedSpan);
+
+    /**
+     * Called after sending a batch of spans.
+     *
+     * @param durationInMillis the time it took to send the spans in milliseconds.
+     * @param batchSize the size of the spans batch that was sent.
+     * @param failed true, if sending has resulted in a failure.
+     */
+    void onSpansSent(long durationInMillis, int batchSize, boolean failed);
```

Comment:
for callbacks that are intended for metrics, probably good to think about an example of how these would be exported. Ex. this particular callback seems cluttered with mixed signals. This would be easy to see when we try to explain how to map this to a timeseries metric.